File: orar/src/main/java/ro/facultate/orar/Entity/Student.java
package ro.facultate.orar.Entity;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "student")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "student_number")
    private Integer studentNumber;

    @ManyToOne
    @JoinColumn(name = "person_id")
    private Person person;

    @ManyToOne
    @JoinColumn(name = "group_id")
    private Group group;
}

File: Orar_Frontend/src/api/models/student/Student.ts
export interface Person {
  id: number;
  firstName: string;
  lastName: string;
  personalId?: string;
  fatherInitial?: string;
}

export interface Student {
  id: number;
  studentNumber: number;
  person: Person;
  group?: {
    id: number;
  };
}